baseline/client/n-back/n-back.js

```diff
@@ -99,7 +103,31 @@ export class NBack {
         return this.constructor.taskName;
     }
 
-    randTrialGroup(n) {
+    randShortPracticeLoop(n) {
+        const passingPracticeData = data => {
+            const noMiss = data.missedIndices.length === 0;
+            const noIncorrect = data.responses.every(r => r.correct);
+            return noMiss && noIncorrect;
+        };
+        return {
+            timeline: [
+                ...this.randTrialGroup(n, 3, 1, false),
+                {
+                    timeline: [this.constructor.simpleInstruction(train_instruction_retry_html)],
+                    conditional_function: () => {
+                        const trialData = jsPsych.data.get().filter({trial_type: "n-back"}).values()[0];
```

Comment:
This doesn't work as expected. See if you can write a test that makes it break. (Hint: Try a test where the user gets it wrong on the first try and right on the second try.)

Reply:
Oh, right! Some for reason, I thought that new data would pushed to the front. I really need more tests in `n-back.test.js`.